metrics/record_test.go

```diff
@@ -226,3 +226,46 @@ func TestMeter(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkMetricsRecording(b *testing.B) {
+	ctx := context.Background()
```

Comment:
Can we add some tags etc. to this? AFAIK that's causing quite some allocations as well.